src/schema/resolvers.ts

```diff
@@ -1,17 +1,31 @@
+import { AppDataSource } from '../data-source.js';
+import { User } from '../entity/User.js';
+
 const resolvers = {
   Query: {
     hello: () => 'Hello, World!',
   },
   Mutation: {
-    createUser: (_, { data }): User => {
-      const id = 1;
-      return { id, ...data };
+    createUser: async (_, { data }): Promise<OutUser> => {
+      console.log('Inserting a new user into the database...');
+      const user = new User();
+      user.name = data.name;
+      user.email = data.email;
+      user.birthDate = data.birthDate;
+      user.password = data.password;
+      await AppDataSource.manager.save(user);
```

Comment:
```suggestion
      const storedUser = await AppDataSource.manager.save(user);
```
se eu não me engano, o save ja retorna o usuario criado

Reply:
Sugestão da sugestão então, já retorna a promise:

```ts
return await AppDataSource.manager.save(user);
```